### DISCUSSION TITLE
osu!taiko "strong" hit sounds don't play unless hitting correctly

### DISCUSSION THREAD

**Discussion by kivkivov** [ppy/osu#25026] [Ideas]
In Stable, finisher hitsounds only play when you hit a finisher note, regardless of if you hit both keys at once or not. In Lazer, this was changed so that finisher hitsounds play **any time** you hit the same color keys at the same time, regardless of what is on the playfield. I find this problematic for the following reasons:

- The finisher hitsounds can play while hitting non-finisher notes, which is very distracting. I'm not sure if it can happen when hitting notes in the 300 window, but when I hit 1/8 or 1/6 notes a little fast, it sometimes plays the finisher hitsounds.
- Many players only hit finishers with one key, so they won't hear the finisher hitsounds. The most common reason is because it does not affect pp. Another reason I experience is that I don't want to achieve a full combo with better accuracy but then not receive pp due to hitting less finishers. For these players, not hearing finisher hitsounds changes how they experience maps on Lazer and makes these maps less satisfying and enjoyable.

Overall, I believe this is a detrimental change and should be reverted. Or at the very least, the player should have the ability to toggle between the two hitsound styles.

**Comment by peppy** [comment]
>  but when I hit 1/8 or 1/6 notes a little fast, it sometimes plays the finisher hitsounds.

Could you provide a recording or replay showing this for reference? We may need to tweak timings.

> Many players only hit finishers with one key, so they won't hear the finisher hitsounds. 

Yep, this was an intentional decision. In osu! we value input-output mapping, so hitting the correct keys is detrimental to getting the full sound effect. We might add support for the old behaviour as a legacy toggle if there's enough people wanting it, but we'll see.

**Reply by hongaaronc** [reply]
Sorry, this isn't a replay/recording, but here's an example of a map where it's very difficult to play this without triggering the finisher sound unintentionally.

https://osu.ppy.sh/beatmapsets/2059491

Not ranked, but this map has frequent `1/12`-snapped doublets, which is a pretty commonly used element in taiko mapping.

So unintentional finisher sounds are hard to avoid at places like: `00:05:852, 00:06:779, 00:07:707, etc...`
...but conversely they won't play at places where these doublets are not monocolor, like: `00:09:563, 00:10:027, 00:10:491, etc...`, which makes the hitsounding feel pretty inconsistent.

**Reply by kivkivov** [reply]
https://osu.ppy.sh/beatmapsets/1869818#taiko/3847138
In this map, the 1/12 notes at about 1:10 hitsound as finishers in autoplay, as an example. And there are many other maps with very fast mono color bursts that would do the same.

Even if you tweak the timings, I still feel it is very jarring and disorienting gameplay-wise to hear the loud, echoing finisher sound any time you accidentally hit the keys together at once. I understand that you want hitsounding to correspond with the player's inputs, but this is a unique problem that no other mode has. Because hitsounds usually don't change depending on how you choose to input them; they simply play or don't play. As a mapper, I personally view finishers as hitsound additions that can be used to auditorily represent the music better. There will always be people who can't hit finishers, [or maps that are very hard for the average player to hit the finishers.](https://osu.ppy.sh/beatmapsets/1173557#taiko/2447689) But from a mapping standpoint, I would like my hitsounding to always be represented, even when a player singletaps the finishers.

**Reply by peppy** [reply]
Subjective opinion. You could also argue that mania is similar. And you could argue this makes sense because in taiko hitsounds are coming from the drum, not the objects. The objects are simply a guide. This is why in taiko (and mania) you can create sounds when there's no objects on screen.

Thanks for the opinions and beatmap links though! We'll factor this into potential future changes. At very least we'll adjust timings.

**Reply by kivkivov** [reply]
Thank you. Here's a video I was just sent of the finisher sounds playing on 1/8 notes.
https://www.youtube.com/watch?v=8pG3Hg9NrBU

**Comment by Alchyr** [comment]
The problem is that finishers are used quite rarely and only to emphasize very specific sounds; having them play/not play at the intended times has a very large impact.

As a taiko mapper, I agree that it would be generally detrimental to the gameplay experience, whether it be a player accidentally hitting two keys at once and getting a finisher sound, or hitting one key for an actual finisher and getting a normal hitsound (less concerned about the second one, but it's still not great; there are maps where the finishers are only theoretically playable and most people will *not* hit them).

**Reply by Alchyr** [reply]
To be more specific, I don't see any upsides to having finisher sounds tied to input; it allows playing finisher sounds when hitting no objects which I'd say is neither an upside or a downside, but during normal gameplay it can only be detrimental to have the wrong sound play when it's a sound as loud/with as much emphasis as a finisher sound has.

**Comment by ulkokos** [comment]
If someone asks themselves why no one is playing taiko in lazer, this is the reason why. This feature is extremely detrimental to the gameplay at higher levels, hitting notes of the same color at a high speed is rather common in this gamemode and the possibility of triggering a Finisher sound makes difficult taiko maps almost impossible to go through.

Take for instance the monocolor 1/8 or 1/12 (or even higher) doublet element, which would look like kk (blue+blue) or dd (red+red), in this case players often will press the keys at a very fast pace and this will trigger the sound out of nowhere. Here's an example I recorded playing https://osu.ppy.sh/beatmapsets/1198368#taiko/2495831 because of it's particular doublet usage and I took the time to do it with both argon pro and the skin I regularily use to play taiko (and that's because I can't even distinguish what's happening with the argon pro hitsounds):
argon pro: https://www.youtube.com/watch?v=ZJ8CVk-hMt4
taiko skin: https://www.youtube.com/watch?v=s__7GSVuHJA

I'm convinced that this feature needs to change asap, and personally I think it should be completely removed. But that's my own personal subjective opinion take, perhaps as peppy said, changing the timings of this feature may solve it but idk man does it? i'm pretty sure it would be better if the sound came from the very object and not your input.

**Reply by ulkokos** [reply]
I'm here with more evidence of this issue, because I want to show that this can happen even with official content and not just graveyarded maps. In this case I was going through the osu!taiko Beatmap Spotlights (Autumn 2023): Playlist C when this happened:

1: https://youtu.be/WwX7NpH8Kr4 (around 0:30 or so)
First example is a video of me playing https://osu.ppy.sh/beatmapsets/1927372#taiko/3980066 this ranked map which has a lot of doublets: you can see that a lot of my inputs end up sounding like finishers very inconsistently.
I'd also like to mention that spinners show in the screen all the time unlike Stable, so in this replay you can see that annoying slow spinner almost the whole map, but that's another issue on its own.

2: https://youtu.be/fRCghaTwWBE
Second example is https://osu.ppy.sh/beatmapsets/1927081#taiko/3979308 and this one is particular because if you pay attention to 0:37 you will see that I got the finisher sound during a 1/6+1/4 pattern (which looks like dddk k dddk), this is the most concerning to me because this didn't happen on a doublet but a whole different pattern that I was just trying to quickly hit.

I may as well mention that the same thing happened to other maps from the very same playlist...

Hope this makes things even more clear about this issue, all of this is very strange because if you change the timings of the finisher sound triggering then where is the line between a simultaneous double click (to hit a finisher) and two separated clicks (to hit a doublet)? imo there's no way to determine which one should trigger such sound. I already said this but this should be removed, I don't think this sets a good precedent for the future of this gamemode.

tldr; ppy pls fix!

**Comment by wwawaawwa** [comment]
I know this post is a few years old but I thought I would give some input. I started playing taiko fairly recently and I had to skin out the finisher sound completely because I kept hearing it in random places. Anything high bpm is painful when there is a random finisher sound in the middle of a pattern I otherwise hit perfectly. I would LOVE to keep the sounds on but they are so distracting in their current state. Maybe only play the sound when a finisher hit actually records bonus points?

**Comment by kivkivov** [comment]
Thank you for bringing attention to this again @wwawaawwa

It has been 2.5 years since my original post, and I am still strongly disappointed with this change. Hitting 1/8 notes slightly fast should not negate the intended hitsounding created by the mapper, this completely sours the beatmap's ability to represent the song. In the dedicated taiko community, this is not a popular change. The 15 upvotes on this thread may as well be 150 with how much smaller (and monolingual Japanese) the taiko playerbase is compared to standard. We just want to be able to experience maps the same way on lazer as we have for the thousands of hours we've poured into the game, yet we can't.

And to provide yet another example, here's a Discord conversation featuring prominent mappers/BNs.

<img width="1001" height="986" alt="3o2ijrio32jrio32r" src="https://github.com/user-attachments/assets/3fa58e57-9610-41d9-ab11-4c1bdea30e87" />


@peppy

**Reply by klariionn** [reply]
Agree with this entirely. The hitsounds alone make it really hard for me to want to play anything taiko related on lazer, alongside other issues I have with how taiko is. I would say this is the most prominent one for me however, since anytime the finisher hitsound pops up during fast rhythms, I lose any sort of timing I had and have to tank score just because of it.

**Reply by JarvisGaming** [reply]
Also agree. The finisher hitsounds playing at random times is really infuriating when trying to play high SR maps on lazer.

**Reply by kivkivov** [reply]
documenting more support from top players and BNs
<img width="806" height="677" alt="Screenshot 2026-04-22 115340" src="https://github.com/user-attachments/assets/99e49371-613d-4047-835c-a39a82ba22e4" />
<img width="660" height="717" alt="Screenshot 2026-04-22 115045" src="https://github.com/user-attachments/assets/f30878ba-3251-4aff-9ca6-95b381923493" />

**Reply by HexaMaster** [reply]
Since the screenshot of my message is here, I'll put the short 16 second clip here of my replay that I mentioned (second kiai only).
https://youtu.be/Hp86tmRLUB4

Hopefully this is good enough to show why most of the really high bpm speed stuff I play is way worse on lazer (if not, just go into lazer and look at the replay of any of the other speed scores in my top plays). And FYI, even though my acc is worse than most other top players which makes me experience more finisher hitsound spam, this happens on every DT/NC play on this map's leaderboard, not just mine. Being spammed with annoying finisher hitsounds is basically unavoidable at this speed, and if you don't have ultra taiko god acc, it makes lazer way more annoying to deal with than stable for anything even moderately fast.

**Comment by peppy** [comment]
Sure we'll revert everything.
@peppy @peppy @peppy

**Reply by kivkivov** [reply]
sorry if the ping was inappropriate
